app/src/main/java/com/rkm/tasky/network/interceptor/TaskyApiKeyInterceptor.kt

```diff
@@ -0,0 +1,19 @@
+package com.rkm.tasky.network.interceptor
+
+import com.rkm.tasky.BuildConfig
+import okhttp3.Interceptor
+import okhttp3.Response
+import javax.inject.Inject
+
+class TaskyApiKeyInterceptor @Inject constructor(): Interceptor {
+
+    private val apiHeader = "x-api-key"
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val newRequest = request.newBuilder().addHeader(
+            apiHeader, BuildConfig.API_KEY
+        )
```

Comment:
I personally got used to adding headers with `header` instead of `addHeader` since `addHeader` will break when you add the same type of header twice (happens more quickly than you think when having multiple interceptors)